File: chat/src/main/java/org/thoughtcrime/securesms/jobs/JobManagerFactories.java
package org.thoughtcrime.securesms.jobs;

import android.app.Application;

import androidx.annotation.NonNull;

import org.thoughtcrime.securesms.jobmanager.Constraint;
import org.thoughtcrime.securesms.jobmanager.ConstraintObserver;
import org.thoughtcrime.securesms.jobmanager.Job;
import org.thoughtcrime.securesms.jobmanager.impl.ChargingConstraint;
import org.thoughtcrime.securesms.jobmanager.impl.ChargingConstraintObserver;
import org.thoughtcrime.securesms.jobmanager.impl.NetworkConstraint;
import org.thoughtcrime.securesms.jobmanager.impl.NetworkConstraintObserver;

import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public final class JobManagerFactories {

    public static Map<String, Job.Factory> getJobFactories(@NonNull Application application) {
        return new HashMap<>() {{
            put(TestJob.KEY, new TestJob.Factory());
            put(PushTextSendJob.KEY, new PushTextSendJob.Factory());
            put(DownloadAttachmentJob.KEY, new DownloadAttachmentJob.Factory());
            put(PushReadReceiptSendJob.KEY, new PushReadReceiptSendJob.Factory());
        }};
    }

    public static Map<String, Constraint.Factory> getConstraintFactories(@NonNull Application application) {
        return new HashMap<>() {{
            put(ChargingConstraint.KEY, new ChargingConstraint.Factory());
            put(NetworkConstraint.KEY, new NetworkConstraint.Factory(application));
        }};
    }

    public static List<ConstraintObserver> getConstraintObservers(@NonNull Application application) {
        return Arrays.asList(new ChargingConstraintObserver(application),
                new NetworkConstraintObserver(application));
    }

}

File: chat/src/main/java/com/difft/android/chat/jobs/JobManagerFactories.kt
package com.difft.android.chat.jobs

import android.app.Application
import com.difft.android.chat.jobmanager.Constraint
import com.difft.android.chat.jobmanager.ConstraintObserver
import com.difft.android.chat.jobmanager.Job
import com.difft.android.chat.jobmanager.impl.NetworkConstraint
import com.difft.android.chat.jobmanager.impl.NetworkConstraintObserver

object JobManagerFactories {

    @JvmStatic
    fun getJobFactories(application: Application): Map<String, Job.Factory<*>> = mapOf(
        PushTextSendJob.KEY to PushTextSendJob.Factory(),
        PushReactionSendJob.KEY to PushReactionSendJob.Factory(),
        DownloadAttachmentJob.KEY to DownloadAttachmentJob.Factory(),
        PushReadReceiptSendJob.KEY to PushReadReceiptSendJob.Factory(),
        PushGroupKeySendJob.KEY to PushGroupKeySendJob.Factory(),
        PushForwardNoticeSendJob.KEY to PushForwardNoticeSendJob.Factory()
    )

    @JvmStatic
    fun getConstraintFactories(application: Application): Map<String, Constraint.Factory<*>> = mapOf(
        NetworkConstraint.KEY to NetworkConstraint.Factory(application)
    )

    @JvmStatic
    fun getConstraintObservers(application: Application): List<ConstraintObserver> = listOf(
        NetworkConstraintObserver(application)
    )
}
